code/game/objects/items/devices/flashlight.dm

```diff
@@ -269,3 +269,51 @@
+
+/obj/item/device/flashlight/emp/Destroy() //If it no longer exists, it will no longer be processed to prevent unnecessary lag
+	processing_objects.Remove(src) //Removes from list of processed items
+	..() //Do the rest of the destroy process
+
+/obj/item/device/flashlight/emp/process() //EMP flashlight process
+	charge_tick++ //Post-increment charge_tick. It increases by 1 every time it is processed.
+	if(charge_tick < 15) //15 ticks required until you gain a flashlight charge
+		return 0 //If it's not 15 ticks then cancel the process until it is called again next tick
+	charge_tick = 0 //If it's 15 ticks reset to 0
+	charge_current = min(charge_current+1, charge_max) //Either add +1 to charge_current (give it another charge in our case), or remain at the value determined by charge_max, depends on which value is lower
+	return 1 //The process is a success and will be called for again next time.
+
+/obj/item/device/flashlight/emp/afterattack(atom/movable/A, mob/user, proximity) //Can use it on anyone and anything as long as you are near them
+	. = ..() //I don't really grasp the idea of what this does but it's important
+	if(!proximity) //If you are not near whatever you use this on...
+		return 0 //Cancel the whole thing
+	if (charge_current > 0) //If you don't have 0 "current charge"
+		charge_current -= 1 //Reduce the charge counter by 1
+		if(ismob(A)) //If whatever you attack is a person
+			var/mob/M = A //Makes A count as M
+			log_attack("<font color='red'>[user.name] ([user.ckey]) has used an EMP flashlight on [M.name] ([M.ckey])!</font>") //Admin logs when checking someone's attack logs
+			M.visible_message("<span class='danger'>[user] has blasted [A] with a pulse! </span>", \
+			"<span class='userdanger'>You have been blasted with a pulse!</span>") //What other people see when someone blasts someone with EMP and what you see if someone blasts you, respectively
+		else //If whatever you attack is not a person
+			log_attack("<font color='red'>[user.name] ([user.ckey]) has used an EMP flashlight on [A.name]!</font>") //Admin logs show what item you used the EMP flashlight on
+			A.visible_message("<span class='danger'>[user] has blasted [A]!</span>") //What people see when someone EMPs an object
+		to_chat(user, "The EMP flashlight has [charge_current] charges left.") //Shows the user how many charges are left on the EMP.
+		A.emp_act(2) //Light EMP pulse
+	else //If you are in proximity but there are no charges
+		to_chat (user, "<span class='warning'>\The [src] must take time to recharge.</span>") //Wait for the EMP flashlight to recharge
+	return //That's it.
+
+/obj/item/device/flashlight/emp/examine(mob/user) //What happens if you examine
+	..() //Examine is normal except for the to_chat appearing afterwards
+	to_chat(user, "Charges: <font color='red'>[charge_current]/4</font>") //Shows you in red how many charges are left out of how many
```

Comment:
Would be useful to only show this if you're very close.

Reply:
Tried to use proximity but not sure where to put it, didn't even give a description. I'll try again

Reply:
`if (get_dist(user, src) <=2)`

Reply:
Thanks for the holder code line